### PR TITLE
Fix exit code

### PR DESCRIPTION
**Pull request by piotmag769**
## Introduced changes

- Extracted `TestFileSummary` to a separate file

## Breaking changes

- Return non-zero exit code when any tests fail

## Checklist

- [x] Linked relevant issue
- [ ] Updated relevant documentation
- [x] Added relevant tests
- [x] Performed self-review of the code
- [ ] Added changes to `CHANGELOG.md`

### CODE REVIEW

File: crates/forge/src/main.rs [hunk 1]
```diff
@@ -68,12 +79,10 @@ fn main_execution() -> Result<()> {
         .output()
         .context("Failed to build contracts with Scarb")?;
     if !build_output.status.success() {
-        bail!(
-            "Scarb build didn't succeed:\n\n{}",
-            String::from_utf8_lossy(&build_output.stdout)
```
Comment (Arcticae) [inline crates/forge/src/main.rs:73]: Why is this removed?

Comment (piotmag769) [reply crates/forge/src/main.rs:73]: Nope, we redirect both stdout and stderr, that's why I removed it

**Review by Arcticae — APPROVED** [review]
